### PR TITLE
added --samplesNames

### PR DESCRIPTION
**Pull request by fidelram**

### CODE REVIEW

**Comment by fidelram** [comment]
@bgruening please check.

File: deeptools/heatmapper.py [hunk 1]
```diff
@@ -562,7 +562,7 @@ def matrix_from_dict(self, matrixDict, regionsDict, parameters):
         self.lengthDict = OrderedDict()
         self.matrixAvgsDict = OrderedDict()
 
-    def read_matrix_file(self, matrix_file, verbose=None, default_group_name='label_1'):
+    def read_matrix_file(self, matrix_file):
```
Comment (fidelram) [inline deeptools/heatmapper.py]: Those parameters were not being used in the function.

File: deeptools/test/test_heatmapper.py [hunk 2]
```diff
@@ -100,14 +100,15 @@ def test_plotHeatmap_scale_regions(self):
 
     def test_plotHeatmap_multi_bigwig_pergroup(self):
         if self.run_image_tests:
-            args = "-m {}/master_multi.mat.gz --perGroup --outFileName /tmp/_test.svg".format(ROOT).split()
+            args = "-m {}/master_multi.mat.gz --perGroup --samplesLabel file1 file2 file3 file4 " \
+                   "--outFileName /tmp/_test.svg".format(ROOT).split()
```
Comment (fidelram) [inline deeptools/test/test_heatmapper.py]: Added a test for `--samplesLabel` that was missing

**Comment by fidelram** [comment]
I ended up changing lot of different files after realizing that the input for `--samplesLabel` was separated by spaces while for `--regionsLabel` was separated by commas. Now both are separated by space.

This PR should also solve the problem in galaxy for plotProfile and clustering that requires that the regionsLabel is the same size as the number of clusters. The default values cause an error. Thus, the default now is 'no value'.

@bgruening would not be better to add a macro for the regions and samples label? that macro can also include plotTitle that is also common between plotHeatmap and plotProfile.

**Comment by vivekbhr** [comment]
Just want to add that there is is --labels option in plotCorrelation, and --sampleLabels in plotHeatmap, which both do the same thing. Maybe we should have one consistent name..

**Comment by fidelram** [comment]
@vivekbhr the difference with plotHeatmap and plotProfile is that there are labels for regions and for samples, hence the distinction.

File: galaxy/wrapper/plotProfiler.xml [hunk 3]
```diff
@@ -120,10 +126,21 @@
                     <option value="se">add standard error</option>
                     <option value="overlapped_lines">overlapped lines</option>
                 </param>
-                <param argument="--regionsLabel" type="text" value="coverage" size="30"
+                <param argument="--samplesLabel" type="text" size="30"
+                    label="Labels for the samples (each bigwig) plotted"
+                    help="The default is to use the file name of the sample. The sample labels should be separated by
+                    spaces and quoted if a label itself contains a space E.g. label-1 &quot;label 2&quot;" />
```
Comment (bgruening) [inline galaxy/wrapper/plotProfiler.xml]: e.g.

**Comment by bgruening** [comment]
:+1:

File: galaxy/wrapper/plotProfiler.xml (hunk 3 above)
Comment (dpryan79) [inline galaxy/wrapper/plotProfiler.xml]: ", e.g., " to be more explicit. Good catch btw!

**Pull request merged**